File: 3159_FindOccurencesofanElementinanArray/3159_FindOccurencesofanElementinanArray.py
class Solution:
    def occurrencesOfElement(self, nums: List[int], queries: List[int], x: int) -> List[int]:
        targetLocations, answers = [], []
        for i in range(len(nums)):
            if nums[i] == x:
                targetLocations.append(i)
        for i in range(len(queries)):
            if queries[i] - 1 < len(targetLocations):
                answers.append(targetLocations[queries[i] - 1])
            else: answers.append(-1)
        return answers
    



    """
     for i, num in enumerate(nums):
            if num == x: 
                arr.append(i)
 
        for q in queries:
            if len(arr) < q: ans.append(-1)
            else: ans.append(arr[q-1])

    works as well        
    """

File: 3159_FindOccurencesofanElementinanArray/3159_FindOccurencesofanElementinanArray.js
/**
 * @param {number[]} nums
 * @param {number[]} queries
 * @param {number} x
 * @return {number[]}
 */
var occurrencesOfElement = function (nums, queries, x) {
  let ans = [];
  let cache = [];
  for (let i = 0; i < nums.length; i++) {
    if (nums[i] == x) cache.push(i);
  }
  for (let i = 0; i < queries.length; i++) {
    queries[i] - 1 < cache.length
      ? ans.push(cache[queries[i] - 1])
      : ans.push(-1);
  }
  return ans;
};
